src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -85,7 +85,19 @@
             librariesUnparsed.put(parsed[0], library);
         }
         List<Addition> additions = new ArrayList<>();
-        LibrariesAction action = build.getAction(LibrariesAction.class);
+        LibrariesAction action = null;
+        
+        for( LibrariesAction laction : build.getActions( LibrariesAction.class ) ) {
+            if( laction.getScope() != null && laction.getScope().equals( scope ) ) {
```

Comment:
Previously, as there was only ever one `LibrariesAction` per build, if it was present, it was concluded that this was a resumed build and that action could be used.

Now, there may be multiple actions so we must find the one whose `scope` equals the `scope` passed into this `add` invocation.